### ISSUE TITLE
Supabase storage bucket using for File storage

### ISSUE THREAD

**Issue by Yeasir0032** [Yeasir0032/Discord-Clone#10]
### Functionality to use supabase storage 
Currently the project uses **uploadthing** for storing files. The new mechanism involves using **supabase's storage** system to store files.
Steps to follow:
1. Read about **supabase storage** system.
2. Understand how supabase works in this project ( for example you can look into this code snippet [page.tsx](https://github.com/Yeasir0032/Discord-Clone/blob/main/app/(setup)/page.tsx)
3. Trace the references of **upload thing**. 
     - It is used when server is created or edited
     - For **file attachment** in messaging
4. Replace the functionality from **upload thing** to **supabase storage** .
5. Make sure to remove uploadthing from **package.json** by uninstalling it.

**Comment by ArtiGaund** [comment]
Can do updating function to supabase storage ? Can You assign this issue to me?

**Comment by Yeasir0032** [comment]
Could you please provide steps how would you achieve this.

**Comment by ArtiGaund** [comment]
1) Initialize supabase client (/config/supabase.js)
import { createClient } from "@supabase/supabase-js"

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL 
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

export default supabase

2) Replacing uploads  from uploadthings folder, Uploading with Supabase storage
import { supabase } from "./config/supabase
async function uploadFile(file){
   const filePath=`...${file.name}`
   const {data,error} = await supabase.storage.from('bucket-name').upload(filePath)
   if(error) console.error("Error while uploading file", error)

    return data
} 
3) Replacing downloads from uploadthings folder, with supabase storage (same as uploads)

4) removing uploading things from package.json.

**Comment by codebreaker3008** [comment]
Can you assign this to me ?

**Comment by codebreaker3008** [comment]
@Yeasir0032 can you assign this issue to me ??

**Comment by Yeasir0032** [comment]
@codebreaker3008 
I have already assigned her.
If she fails to do so then you will be assigned

**Comment by ArtiGaund** [comment]
@Yeasir0032 Is there any timeline to complete it?

**Comment by Yeasir0032** [comment]
This issue will not take more than 4 days. 
If you need some more you can ask for it but if become inactive then I have to give chance to others

**Comment by ArtiGaund** [comment]
I will do it try to complete it  till tomorrow

**Comment by ArtiGaund** [comment]
I have to remove everything from the whole project? wherever it is used?

**Comment by ArtiGaund** [comment]
![Screenshot (267)](https://github.com/Yeasir0032/Discord-Clone/assets/38074085/665f18ee-1318-4c8f-b4f8-05b236088c41)
![Screenshot (268)](https://github.com/Yeasir0032/Discord-Clone/assets/38074085/7167e5e4-1287-425a-aaca-6ee1aaaeb02a)
![Screenshot (269)](https://github.com/Yeasir0032/Discord-Clone/assets/38074085/20fb6ff7-d85d-4805-ad09-5f489d772b67)
![Screenshot (270)](https://github.com/Yeasir0032/Discord-Clone/assets/38074085/8c22f7cc-9a95-4591-9e13-baa259a53fb2)
![Screenshot (271)](https://github.com/Yeasir0032/Discord-Clone/assets/38074085/e48ca66b-30ad-48bb-9804-64741c91d253)
![Screenshot (272)](https://github.com/Yeasir0032/Discord-Clone/assets/38074085/d83ea1c4-41a4-401f-923e-0a4b4ab1248f)
Are these correct?

**Comment by codebreaker3008** [comment]
@Yeasir0032  could you assign this to me too, accept the one which does it earlier

**Comment by ArtiGaund** [comment]
@Yeasir0032 Where are all the table names and their fields of  Supabase database?

**Comment by ArtiGaund** [comment]
@Yeasir0032 Where are the tables of Supabase with their fields and their datatype?

**Comment by Yeasir0032** [comment]
![Screenshot from 2024-05-11 20-28-26](https://github.com/Yeasir0032/Discord-Clone/assets/124646897/0b25d183-ad93-4ac0-ba0d-e106312d402d)
Here is the schema visualizer

**Comment by growupanand** [comment]
I am working on the file preview component #19, How much time it can take for a new upload flow any idea?

**Comment by ArtiGaund** [comment]
You can assign this issue to someone. My setup is not working, and it is not good to hold an assignment if I am not able to work on it.

**Comment by Yeasir0032** [comment]
@codebreaker3008 
Do you want to do it?

**Comment by codebreaker3008** [comment]
@Yeasir0032  yes but i am caught up in some personal issues, you could assign it to me but it will take some time

**Comment by Yeasir0032** [comment]
Hey @ArtiGaund , could you update your progress. Let's fix the **error** you are facing.

**Comment by ArtiGaund** [comment]
Created buckets in database done

**Comment by PayalKumari10** [comment]
Hello, @ArtiGaund  any update on this issue?

**Comment by ArtiGaund** [comment]
This issue is not assigned to me.
